From: Luca Weiss <luca.weiss@fairphone.com>
Date: Thu, 28 Sep 2023 11:48:56 +0200
Subject: arm64: dts: qcom: qcm6490-fairphone-fp5: Add OCP96011 audio switch

Add a node for the OCP96011 on the board which is used to handle USB-C
analog audio switch and handles the SBU mux for DisplayPort-over-USB-C.

Signed-off-by: Luca Weiss <luca.weiss@fairphone.com>
---
 arch/arm64/boot/dts/qcom/qcm6490-fairphone-fp5.dts | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/qcom/qcm6490-fairphone-fp5.dts b/arch/arm64/boot/dts/qcom/qcm6490-fairphone-fp5.dts
index 809ec036..bcdf4fc 100644
--- a/arch/arm64/boot/dts/qcom/qcm6490-fairphone-fp5.dts
+++ b/arch/arm64/boot/dts/qcom/qcm6490-fairphone-fp5.dts
@@ -694,8 +694,16 @@ vreg_l7p: ldo7 {
 		};
 	};
 
+	typec-mux@42 {
+		compatible = "ocs,ocp96011", "fcs,fsa4480";
+		reg = <0x42>;
+
+		interrupts-extended = <&tlmm 7 IRQ_TYPE_LEVEL_LOW>;
+
+		vcc-supply = <&vreg_bob>;
+	};
+
 	/* Pixelworks @ 26 */
-	/* FSA4480 USB audio switch @ 42 */
 	/* AW86927FCR haptics @ 5a */
 };
 
